File: jan_contest/prob_3/guess.py
def same_checker(i, j, entire_list, animals):
    count = 0
    a1 = entire_list[animals[i]]
    a2 = entire_list[animals[j]]

    for k in range(len(a1)):
        if a1[k] in a2:
            count+=1
    return count
    
entire_list = dict()
animals = []

with open("guess.in" , "r") as fin:
    numb_ani = int(fin.readline().strip())
    
    for i in range(numb_ani):
        list_arr = []
        
        line = fin.readline().strip().split(" ")
        numb_of_chars = int(line[1])
        
        for j in range(numb_of_chars):
            list_arr.append(line[2+j])
        entire_list[line[0]] = list_arr
        animals.append(line[0])
            
    largest = 0
    for i in range(numb_ani):
        for j in range(i+1, numb_ani):
            largest = max(largest, same_checker(i, j, entire_list, animals))
            
with open("guess.out", "w") as fout:
    fout.write(str(largest+1))

File: jan_contest/prob_3/guess.cpp
#include <iostream>
#include <vector>
#include <fstream>
using namespace std;

vector<string> chars[100];
int num_counter(int i, int j) {
    vector<string> &a1 = chars[i], &a2 = chars[j];
    
    int counter = 0;
    for(int i = 0; i < a1.size(); i++){
        for(int j = 0; j < a2.size(); j++){
            if(a1[i] == a2[j]) {
                counter++;
            }
        }
    }
    return counter;
}


int main() {
    ifstream fin ('guess.in')
    int numb_ani;
    int numb_of_chars;
    string animal;
    fin >> numb_ani;

    for(int i = 0; i < numb_ani; i++) {
        list_arr = [];

        fin >> animal >> numb_of_chars;
    } 
}